### ISSUE TITLE
Suggestion: Bind to center of large tokens

### ISSUE THREAD

**Issue by jonathanmclaus** [KayelGee/token-attacher#103]
Elements attached to a token that occupies multiple squares treat the upper left square as the attach point, meaning rotating a larger token does not properly rotate the elements attached. Instead the center of the token should be considered.

**Comment by Tarrith** [comment]
I second this request.

**Comment by KayelGee** [comment]
I don't understand the issue.
I've tested both a small token attached to a bigger token and vice versa. Both rotate correctly for me.
Can you give me a step by step instruction to replicate what you mean? Or send me a prefab that shows this issue.

**Comment by Tarrith** [comment]
I have the feeling the issue only arises if a token has been given some width.

![image](https://github.com/KayelGee/token-attacher/assets/23333805/a5a68adc-0558-4149-902d-83596fb89e41)

**Comment by KayelGee** [comment]
Still can't replicate:
[rotate_wide.webm](https://github.com/KayelGee/token-attacher/assets/20637397/461a40a7-401a-496f-938d-0d70a33472d1)

**Comment by Tarrith** [comment]
Hex grid with the flats on the top?

**Comment by KayelGee** [comment]
ohhh hex grids. Yeah there's no special code in token attacher for hex grids, so the math probably doesn't work out.
Is it only tokens that have the problem?
The program I used is screentogif.

**Comment by Tarrith** [comment]
I haven't tried anything else but tokens
...I didn't realise you could attack other things to be honest 😄.
Yeah hex grids seem to be a pain, but it's a bit of a standard I suppose with tactical warfare games, which is what I'm working on space battle wise.

Lol I might have hijacked the original posters that though, oops!

**Comment by KayelGee** [comment]
I have a suspicion that they might've worked with hex grids aswell, as square grid works as it should from my testing

**Comment by Tarrith** [comment]
Yup, you're right. It's definitely only an issue on Hex grids. Just tested it on a square with no issues.

**Comment by jonathanmclaus** [comment]
I had issues with a square grid, but looks like it's not limited to rotation (it's just what I noticed first). In this example, I attach a 1x1 tile (size given relative to the grid, not its actual size) to a 2x2 token, offset by an entire grid square diagonally. As you can see, once transformations are applied, token attacher seems to be using only the top left square of the 2x2 token. It first scales down the tile, and then it rotates it around a single square of the token

![Animation](https://github.com/KayelGee/token-attacher/assets/5731591/a268535d-df7e-41e7-be3a-c2eeeaee6bcb)

**Comment by Tarrith** [comment]
I stand corrected.

I apologise for hijacking your thread as I thought it issues were the same.

**Comment by jonathanmclaus** [comment]
Here's an example where I demonstrate the tile is not attached to begin with. Furthermore, it demonstrates that detaching and reattaching the same tile repeatedly iterates on the problem - the second iteration causes the tile to scale down to 25% in each dimension.

File is too big for Github, but the video can be found here: https://i.imgur.com/SYxJCTI.mp4

**Comment by KayelGee** [comment]
Sorry for getting back to this so late, but while your video shows what you say, I can't replicate it. This is probably some module interaction that breaks token attachers attaching.

**Comment by Tarrith** [comment]
The new foundry broke the game anyway.
I'll try again after they fix stuff.

**Comment by KayelGee** [comment]
I'm still in the process of updating token attacher to v12 so I didn't want to ask if you can replicate it in the current version. Once the update is out you can though.

**Comment by Tarrith** [comment]
Sounds like a plan :)

**Comment by KayelGee** [comment]
I've added a pre-release for V12 if you want to check, if the issue still persists.

**Comment by jonathanmclaus** [comment]
I still have this issue on the latest version of Token Attacher (v4.6.7). When you tested / tried to replicate it, was your token 1x1 or was it larger?

**Comment by Tarrith** [comment]
The larger token was being controlled by the smaller token. So when I rotate the small token the larger token was being rotated but the rotation was being skewed. 
The issue still persists even after creating a totally new scene. 

I can export the scene, hex map, and attach them if that would help.

**Comment by jonathanmclaus** [comment]
To be clear, my issue is with a standard square grid. Maybe the underlying cause is the same, but my issue really has nothing to do with hex grids.
